### DISCUSSION TITLE
Upsert with undefined?

### DISCUSSION THREAD

**Discussion by hatchli** [prisma/prisma#2721] [Q&A] [answered]
Can you upsert with a potentially undefined where variable? I ask because the playground doesn't permit it, but the error coming back from prisma when I use null (which playground does accept) is: 
`Argument id for data.identification.upsert.0.where.id must not be null. Please use undefined instead`
The goal here is to update the id if the id exists, and if it doesn't exist then create one (my schema autoincrements the id by default).
e.g.,
```
{
  "where": {"id": 19},
  "data": {
    "identification": {
      "upsert": {
        "where": {"id": undefined},
        "update": {"type": "Not Passport","id_number": "123","photo": false},
        "create": {"type": "Not Passport","id_number": "123","photo": false}
      }
    }
    }
}
```

**Comment by ryands17** [comment, marked as answer]
Hey @hatchli :wave: 
Could you create a feature request for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=)?

I tried with `undefined` and `null` and it doesn't work. So as a workaround, you would have to pass a value that doesn't exist if the id is null or undefined.

Something like:
```ts
{
  "where": {"id": 19},
  "data": {
    "identification": {
      "upsert": {
        "where": { "id": id === undefined ? -1 : id },
        "update": { "type": "Not Passport","id_number": "123","photo": false },
        "create": { "type": "Not Passport","id_number": "123","photo": false }
      }
    }
    }
}
```

**Reply by mmaazahmed-diya** [reply]
is this still the recommended approach?

**Reply by Huh-David** [reply]
An empty string is also an option (`id: ""`)

**Comment by ludralph** [comment]
Hi @hatchli,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
